components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,41 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuredCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        let type_length = v.len();
+        write_length(self.w, type_length);
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, type_length));
+    }
+}
+
+struct StructuredCloneReader {
+    r: *mut JSStructuredCloneReader,
+}
+
+impl StructuredCloneReader {
+    unsafe fn read_bytes(&self) -> Vec<u8> {
+        let mut type_length = vec![0u8; read_length(self.r)];
```

Comment:
Let's rename this `bytes` instead of `type_length`, since it's a buffer and not a length.